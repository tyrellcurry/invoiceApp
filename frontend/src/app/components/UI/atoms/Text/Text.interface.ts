type TextVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'body' | 'kicker' | 'custom';
export interface ITextProps {
  children: React.ReactNode;
  className?: string;
  tag?: React.ElementType;
  variant?: TextVariants;
}
