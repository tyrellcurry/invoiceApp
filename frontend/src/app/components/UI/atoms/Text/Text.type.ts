type TextVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'body' | 'kicker' | 'custom';
export type TTextProps<T extends React.ElementType> = {
  children: React.ReactNode;
  className?: string;
  tag?: T;
  variant?: TextVariants;
} & React.ComponentPropsWithoutRef<T>;
