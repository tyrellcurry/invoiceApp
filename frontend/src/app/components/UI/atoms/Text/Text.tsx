/**
 * @name Text
 * @author Tyrell Curry <tyrellcurryio@gmail.com>
 *
 * Used for headings and basic text elements
 *
 * @param tag - element's HTML tag
 * @param variant - element's styling
 * @param className - element's custom classes
 * @param children - nested ReactNodes
 *
 * @returns {JSX.Element}
 */

import { JSX } from 'react';
import { ITextProps } from './Text.interface';
import classNames from 'classnames';

const textVariants = {
  h1: 'text--h1',
  h2: 'text--h2',
  h3: 'text--h3',
  h4: 'text--h4',
  h5: 'text--h5',
  body: 'text--body',
  kicker: 'text--kicker',
  custom: 'text--custom',
};

const Text = (props: ITextProps): JSX.Element => {
  const { tag = 'div', variant = 'body', className: propsClassName, children } = props;
  const TagName = tag || 'div';
  const className = classNames(textVariants[variant], propsClassName);
  return <TagName {...{ className }}>{children}</TagName>;
};

export default Text;
