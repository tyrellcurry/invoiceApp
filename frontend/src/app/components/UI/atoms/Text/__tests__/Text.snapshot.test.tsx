import React from 'react';
import { render } from '@testing-library/react';
import Text from '@/app/components/UI/atoms/Text/Text';

describe('Text Component', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<Text>Default Text</Text>);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with a custom tag and variant', () => {
    const { container } = render(
      <Text tag="h1" variant="h1" className="custom-class">
        Heading Text
      </Text>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with kicker variant', () => {
    const { container } = render(<Text variant="kicker">Kicker Text</Text>);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with custom children', () => {
    const { container } = render(<Text>Custom Children Content</Text>);
    expect(container).toMatchSnapshot();
  });
});
