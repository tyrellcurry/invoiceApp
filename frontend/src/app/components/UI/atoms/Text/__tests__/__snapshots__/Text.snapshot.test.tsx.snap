// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders correctly with a custom tag and variant 1`] = `
<div>
  <h1
    class="text text--h1 custom-class"
  >
    Heading Text
  </h1>
</div>
`;

exports[`Text Component renders correctly with custom children 1`] = `
<div>
  <div
    class="text text--body"
  >
    Custom Children Content
  </div>
</div>
`;

exports[`Text Component renders correctly with default props 1`] = `
<div>
  <div
    class="text text--body"
  >
    Default Text
  </div>
</div>
`;

exports[`Text Component renders correctly with kicker variant 1`] = `
<div>
  <div
    class="text text--kicker"
  >
    Kicker Text
  </div>
</div>
`;
